test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,382 @@
+    {
+      std::vector<bool> expectedFilteredMask(inputCloud->size(), false);
+      pcl::Indices expectedRemovedIndices;
+      pcl::PointCloud<pcl::PointXYZ> expectedCloud;
+      pcl::copyPointCloud(*inputCloud, expectedFilteredIndices, expectedCloud);
+      for (pcl::index_t idx : expectedFilteredIndices) {
+        expectedFilteredMask[idx] = true;
+      }
+      for (size_t i = 0; i < inputCloud->size(); ++i) {
+        if (!expectedFilteredMask[i]) {
+          expectedRemovedIndices.push_back(i);
+        }
+      }
+
+      cropHullFilter.setInputCloud(inputCloud);
+
+      pcl::Indices filteredIndices;
+      cropHullFilter.filter(filteredIndices);
+      pcl::test::EXPECT_EQ_VECTORS(expectedFilteredIndices, filteredIndices);
+      pcl::test::EXPECT_EQ_VECTORS(expectedRemovedIndices, *cropHullFilter.getRemovedIndices());
+      // check negative filter functionality
+      {
+        cropHullFilter.setNegative(true);
+        cropHullFilter.filter(filteredIndices);
+        pcl::test::EXPECT_EQ_VECTORS(expectedRemovedIndices, filteredIndices);
+        pcl::test::EXPECT_EQ_VECTORS(expectedFilteredIndices, *cropHullFilter.getRemovedIndices());
+        cropHullFilter.setNegative(false);
+      }
+      // check cropOutside functionality
+      {
+        cropHullFilter.setCropOutside(false);
+        cropHullFilter.filter(filteredIndices);
+        pcl::test::EXPECT_EQ_VECTORS(expectedRemovedIndices, filteredIndices);
+        pcl::test::EXPECT_EQ_VECTORS(expectedFilteredIndices, *cropHullFilter.getRemovedIndices());
+        cropHullFilter.setCropOutside(true);
+      }
+
+      pcl::PointCloud<pcl::PointXYZ> filteredCloud;
+      cropHullFilter.filter(filteredCloud);
+      ASSERT_EQ (expectedCloud.size(), filteredCloud.size());
+      for (pcl::index_t i = 0; i < expectedCloud.size(); ++i)
+      {
+        EXPECT_XYZ_NEAR(expectedCloud[i], filteredCloud[i], 1e-5);
+      }
+      // check non empty out cloud filtering
+      cropHullFilter.filter(filteredCloud);
+      EXPECT_EQ (expectedCloud.size(), filteredCloud.size());
+
+      // check keep organized
+      {
+        cropHullFilter.setKeepOrganized(true);
+        cropHullFilter.setUserFilterValue(-10.);
+        pcl::PointXYZ defaultPoint(-10., -10., -10.);
+        cropHullFilter.filter(filteredCloud);
+        ASSERT_EQ (inputCloud->size(), filteredCloud.size());
+        for (pcl::index_t i = 0; i < inputCloud->size(); ++i)
+        {
+          pcl::PointXYZ expectedPoint = expectedFilteredMask[i] ? inputCloud->at(i) : defaultPoint;
+          EXPECT_XYZ_NEAR(expectedPoint, filteredCloud[i], 1e-5);
+        }
+        cropHullFilter.setKeepOrganized(false);
+      }
+    }
+  };
+
+  bool randomBool() {
+    static auto gen = std::bind(
+        std::uniform_int_distribution<>(0,1),
+        std::default_random_engine());
+    return gen();
+  }
+
+  std::vector<std::pair<pcl::Indices, pcl::PointCloud<pcl::PointXYZ>::ConstPtr>>
+  createTestData(
```

Comment:
(2) This should go in the `SetUp` function of test fixtures. Break down your test functions to simpler and smaller functions

Reply:
broke a complex test into 8 small tests: 8 for 2d case and 8 for 3d case